src/lib.rs

```diff
@@ -533,16 +535,34 @@ pub trait Rng: RngCore {
     /// let mut rng = thread_rng();
     /// println!("{}", rng.gen_bool(1.0 / 3.0));
     /// ```
+    #[inline]
+    fn gen_bool(&mut self, p: f64) -> bool {
+        assert!(0.0 <= p && p <= 1.0);
+        p > self.gen()
+    }
+
+    /// Return a bool with a probability of `numerator/denominator` of being
+    /// true. I.e. `gen_ratio(2, 3)` has chance of 2 in 3, or about 67%, of
+    /// returning true. If `numerator == denominator`, then the returned value
+    /// is guaranteed to be `true`. If `numerator == 0`, then the returned
+    /// value is guaranteed to be `false`.
     ///
     /// # Panics
     ///
-    /// If `p` < 0 or `p` > 1.
+    /// If `denominator == 0` or `numerator > denominator`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use rand::{thread_rng, Rng};
+    ///
+    /// let mut rng = thread_rng();
+    /// println!("{}", rng.gen_ratio(2, 3));
+    /// ```
     ///
-    /// [`distributions::Bernoulli`]: distributions/bernoulli/struct.Bernoulli.html
     #[inline]
-    fn gen_bool(&mut self, p: f64) -> bool {
-        let d = distributions::Bernoulli::new(p);
-        self.sample(d)
+    fn gen_ratio(&mut self, numerator: u32, denominator: u32) -> bool {
+        self.gen_bool(numerator as f64 / denominator as f64)
```

Comment:
Unless there's a good reason for this, I think we should just wrap `Bernoulli::from_ratio` instead.